openlibrary/plugins/upstream/models.py

```diff
@@ -111,6 +113,16 @@ def get_ia_cover(self, itemid, size):
         w, h = image_sizes[size.upper()]
         return "https://archive.org/download/%s/page/cover_w%s_h%s.jpg" % (itemid, w, h)
 
+    def get_isbn13(self):
+        """Fetches either isbn10 or isbn13 from record and returns canonical
+        isbn13
+        """
+        isbn_13 = self.isbn_13 and canonical(self.isbn_13[0]) or ""
+        if not isbn_13:            
+            isbn_10 = self.isbn_10 and self.isbn_10[0] or ""
+            return isbn_10 and canonical(isbn_10_to_isbn_13(isbn_10))
+        return isbn_13
+    
     def get_identifiers(self):
```

Comment:
Does this not do what you need? Then `asin` could be added to the list. Even though this code is simplifying the isbn handling on the html pages, it still looks like the generated isbn13 is being converted back to an isbn10. The model should just return the identifiers, isbn10, isbn13, and asin -- the html shouldn't have so much work to do to figure these out.

Reply:
cc: @cdrini

Reply:
We looked at this and decided for now just to DRY the template as a 1st step assuming we're only every dealing w/ isbn13. Because amazon requires isbn10, we convert back if/when required. Some isbns also have bad formatting and we attempt to address that here.

This PR is for BWB so if we want to have a followup specific to isbn normalization, I support that. But as it is, refactoring isbn logic out of the template was pushing the scope (and the outcome is probably better than what was in the original PR so I don't want to block BWB any longer on getting their prices showing up)

Reply:
It doesn't look like that method does isbn10 to 13 (or vv) conversion :/ I create an issue to try to clean this stuff up: https://github.com/internetarchive/openlibrary/issues/2439 . In the mean time marking as won't fix for now.